ambari-server/src/main/resources/Ambari-DDL-Derby-CREATE.sql

```diff
@@ -1044,6 +1044,28 @@ CREATE TABLE upgrade_history(
   CONSTRAINT UQ_upgrade_hist_srvc_grp UNIQUE (upgrade_id, service_group_id)
 );
 
+CREATE TABLE upgrade_plan(
+  id BIGINT NOT NULL,
+  cluster_id BIGINT NOT NULL,
+  upgrade_type VARCHAR(255) DEFAULT 'ROLLING' NOT NULL,
+  direction VARCHAR(255) DEFAULT 'UPGRADE' NOT NULL,
+  skip_failures SMALLINT DEFAULT 0 NOT NULL, 
+  skip_sc_failures SMALLINT DEFAULT 0 NOT NULL, 
+  skip_prechecks SMALLINT DEFAULT 0 NOT NULL,
+  fail_on_precheck_warnings SMALLINT DEFAULT 0 NOT NULL,
+  skip_service_checks SMALLINT DEFAULT 0 NOT NULL,
+  CONSTRAINT PK_upgrade_plan PRIMARY KEY (id)
```

Comment:
Think of how we're going to enforce the idea of "One upgrade plan per cluster at a time" ... We can do a lot of this in the business logic, but what about adding a constraint here? Once the upgrade has started, we can remove the plan, right? So, should we enforce a cluster ID uniqueness here?

Reply:
We need to discuss this.  If we are taking options from a plan and doing what we do today (with upgrade history) to do a downgrade or revert then maybe we can delete it.  But maybe only after finalize, not start.

Reply:
Yes - maybe something for the next iteration of this work.